### PR TITLE
Add Enterprise license type

### PR DESCRIPTION
**Pull request by ph**
Add the Enterprise license type and regenerate the types files.

Fixes: #14246

### CODE REVIEW

**Comment by ph** [comment]
need to be backported to 7.5, 7.x

File: x-pack/libbeat/licenser/licensetype_string.go [hunk 1]
```diff
@@ -8,9 +8,22 @@ package licenser
 
 import "strconv"
 
-const _LicenseType_name = "Open sourceTrialBasicStandardGoldPlatinum"
+func _() {
+	// An "invalid array index" compiler error signifies that the constant values have changed.
+	// Re-run the stringer command to generate them again.
+	var x [1]struct{}
```
Comment (ph) [inline x-pack/libbeat/licenser/licensetype_string.go]: Oh they changed their method.

**Comment by ph** [comment]
@urso After looking at the code, I see that we are not sending an appropriate `User-Agent` for every HTTP request.

We should send one unique per beats something like `Filebeat v7.5.0`, `Metricbeat v7.5.0`, etc.. Looking at the code, I don't see a good way to retrieve the beat name and the version when the ES client is created without linking the version package (assuming that we also add the name there) and the `elasticsearch` package or doing other refactoring.

The query string options seems less problematic for us.

**Comment by urso** [comment]
@ph A beat can customize its name and even version. You need access to beat.Info in order to create the UA. Maybe we can add the user agent as paramter to elasticsearch.NewClient or its configure structure.

**Comment by ph** [comment]
sound good, we already have what we need to do it.

https://github.com/elastic/beats/blob/fleet/agent/kibana/round_trippers.go#L33-L51

**Comment by ph** [comment]
testing license-related check is not a simple task when its not "trial" or "basic", in the linked private issues there are test servers that @bytebilly has created with credentials to be able to tested with a real server. The other way is to use the generated license in the private repository with dev artifact to run a local server.

**Comment by faec** [comment]
Initial test pass succeeded (beats work fine with an enterprise cluster), but turned up some logging issues that @ph is fixing in #15834: we don't currently log the detected license anywhere (beats only report whether the license check was successful, not which license is in use), and some log messages during the license check were being dropped. Will do another test pass once the fix gets in to confirm that enterprise clusters report the correct license.

**Comment by faec** [comment]
The logging changes above made it into today's build candidate, and another testing pass confirmed that everything looks good

**Comment by ph** [comment]
Thanks @faec :tada:

**Pull request merged**